bin/oneoff/import_unit_details.rb

```diff
@@ -130,17 +143,33 @@ def get_validated_chapters(cb_unit)
     raise "no chapters or lessons found"
   end
 
-  cb_unit['chapters'].presence || [{'lessons' => cb_unit['lessons']}]
+  cb_unit['lessons'].presence || cb_unit['chapters'].map {|ch| ch['lessons']}.flatten
 end
 
-def get_validated_lesson_pairs(script, chapters)
+# Because not all lockable lessons in code studio have lesson plans in
+# Curriculum Builder, there may be a mismatch between the number of lessons in
+# CB and Code Studio. This method does the following:
+# 1. Verifies that every non-lockable lesson in Code Studio has a lesson plan in CB.
+# 2. Verifies that every lesson plan in CB has a corresponding lesson in Code Studio,
+#    including any lesson plans for lockable lessons.
+# 3. Returns a list of pairs of all Code Studio Lessons with corresponding
```

Comment:
I found that we also need to do something with lockable lessons to put the levels in activity sections. I'm not really sure how to represent that in this script though.

Reply:
I see... so ideally, lockable lessons would have a presence in the lesson_pairs list? if that's the direction we want to head in, I could find a different way to do what I am doing here, which doesn't depend on paired_lessons. would that help?

Reply:
@bethanyaconnor since other things are stacking up behind this, and you're out today, I'd like to go ahead and merge. When you're back, I'd like to make sure I understand the issue you're raising and address in a subsequent PR if needed.

Reply:
Yes, sorry, that sounds good!

Reply:
per convo in retro, I will not do any follow-up work here. thanks for working around the way I did this by depending on lesson_pairs 😳